src/change_stream/options.rs

```diff
@@ -0,0 +1,77 @@
+//! Contains options for ChangeStreams.
+use serde::{Deserialize, Serialize};
+use serde_with::skip_serializing_none;
+use std::time::Duration;
+use typed_builder::TypedBuilder;
+
+use crate::{change_stream::event::ResumeToken, collation::Collation, options::AggregateOptions};
+
+#[skip_serializing_none]
+#[derive(Clone, Debug, Default, Deserialize, Serialize, TypedBuilder)]
+#[serde(rename_all = "camelCase")]
+#[non_exhaustive]
+/// These are the valid options that can be passed to the `watch` method for creating a
+/// [`ChangeStream`](../struct.ChangeStream.html).
+pub struct ChangeStreamOptions {
```

Comment:
One thing the original PR did not account for is sessions, which we'll need to do now. I think we should try to mimic the API of cursors here where we have a `ChangeStream` type and a `SessionChangeStream`. We may need to introduce an internal change stream type that both can wrap as part of that.

Reply:
Sessions hadn't occurred to me and the api surface gets ugly with the `_with_session` x `_with_pipeline` matrix - currently stepping back to see if I can figure out a non-terrible way to expose this.

Reply:
Ah yeah, hadn't thought about that. One possible solution is to move the problem to `Cursor` / `ChangeStream` and have a `fn with_type<T>(self) -> Cursor<T>` method. Then it would look like:
```rust
let cs = collection.watch(vec![...], None).await?.with_type::<MyType>();
```
Not super pretty either but it does generalize well.

Reply:
How would that be implemented, though?  `Cursor` has an internal buffer that can't be trivially converted.

Reply:
It just so happens that the ticket I'm working on now is to change the backing buffer of `Cursor` to `RawDocumentBuf`, which would make this change pretty easy. It used to be this way (with `Document` though) but was updated to use `T` as part of the perf work. Now that we have a more expressive raw API, we don't need to deserialize everything up front anymore.

Reply:
Updated the API to expose `ChangeStream::with_type` and added session methods and types.